File: pdu/python/hako_srv_msgs/pdu_cdr_conv_GetEventRequest.py
from .pdu_pytype_GetEventRequest import GetEventRequest
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_GetEventRequest(writer: CdrWriter, src: GetEventRequest):
    writer.write_string(src.name)


def cdr_body_to_py_GetEventRequest(reader: CdrReader, dst: GetEventRequest):
    dst.name = reader.read_string()
    return dst


def py_to_cdr_GetEventRequest(src: GetEventRequest) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_GetEventRequest(writer, src)
    return writer.bytes()


def cdr_to_py_GetEventRequest(cdr_payload) -> GetEventRequest:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = GetEventRequest()
    return cdr_body_to_py_GetEventRequest(reader, dst)

File: pdu/javascript/hako_srv_msgs/pdu_cdr_conv_GetEventRequest.js
import { GetEventRequest } from './pdu_jstype_GetEventRequest.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduGetEventRequestConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { GetEventRequest } src
     */
    static to_cdr_body(writer, src) {
        writer.write_string(src.name);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { GetEventRequest } dst
     * @returns { GetEventRequest }
     */
    static cdr_body_to_js(reader, dst) {
        dst.name = reader.read_string();
        return dst;
    }

    /**
     * @param { GetEventRequest } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { GetEventRequest }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new GetEventRequest());
    }
}
